apm-sniffer/apm-sdk-plugin/cassandra-java-driver-3.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/cassandra/java/driver/v3/SessionManagerExecuteAndExecuteAsyncWithStatementArgInterceptor.java

```diff
@@ -0,0 +1,82 @@
+import org.apache.skywalking.apm.agent.core.context.tag.Tags;
+import org.apache.skywalking.apm.agent.core.context.trace.AbstractSpan;
+import org.apache.skywalking.apm.agent.core.context.trace.SpanLayer;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.EnhancedInstance;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.InstanceMethodsAroundInterceptor;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.MethodInterceptResult;
+import org.apache.skywalking.apm.network.trace.component.ComponentsDefine;
+
+import java.lang.reflect.Method;
+
+/**
+ * @author stone.wlg
+ */
+public class SessionManagerExecuteAndExecuteAsyncWithStatementArgInterceptor implements InstanceMethodsAroundInterceptor {
+
+    @Override
+    public final void beforeMethod(EnhancedInstance objInst, Method method, Object[] allArguments,
+                                   Class<?>[] argumentsTypes,
+                                   MethodInterceptResult result) throws Throwable {
+        ConnectionInfo connectionInfo = (ConnectionInfo) objInst.getSkyWalkingDynamicField();
+        if (connectionInfo == null) {
+            return;
+        }
+
+        Statement statement = (Statement) allArguments[0];
+        String remotePeer = statement.getHost() == null ? connectionInfo.getContactPoints() : (statement.getHost().getSocketAddress().getHostName() + ":" + statement.getHost().getSocketAddress().getPort());
+        String keyspace = statement.getKeyspace() == null ? connectionInfo.getKeyspace() : statement.getKeyspace();
+        String query = statement.toString();
+        if (statement instanceof BoundStatement) {
+            query = ((BoundStatement) statement).preparedStatement().getQueryString();
+        }
+
+        AbstractSpan span = ContextManager.createExitSpan(Constants.CASSANDRA_OP_PREFIX + method.getName(), remotePeer);
+        span.setComponent(ComponentsDefine.CASSANDRA_JAVA_DRIVER);
+        Tags.DB_TYPE.set(span, Constants.CASSANDRA_DB_TYPE);
+        Tags.DB_INSTANCE.set(span, keyspace);
+        Tags.DB_STATEMENT.set(span, query);
+        SpanLayer.asDB(span);
+    }
+
+    @Override
+    public final Object afterMethod(EnhancedInstance objInst, Method method, Object[] allArguments,
+                                    Class<?>[] argumentsTypes,
+                                    Object ret) throws Throwable {
+        ConnectionInfo connectionInfo = (ConnectionInfo) objInst.getSkyWalkingDynamicField();
+        if (connectionInfo != null) {
+            ContextManager.stopSpan();
+        }
+        return ret;
+    }
+
+    @Override
+    public final void handleMethodException(EnhancedInstance objInst, Method method, Object[] allArguments,
+                                            Class<?>[] argumentsTypes, Throwable t) {
+        AbstractSpan span = ContextManager.activeSpan();
+        span.errorOccurred();
+        span.log(t);
```

Comment:
We have to make sure the span is exist and created by this interceptor. So we have to double-check in same conditions(`if (connectionInfo != null) {`). 
And to check `ContenxtManager.isActive()` to make it is be safely. It is the same in others interceptor.

Reply:
Got it, thanks